cloudinit/distros/arch.py

```diff
@@ -43,16 +43,17 @@ def __init__(self, name, cfg, paths):
         cfg['ssh_svcname'] = 'sshd'
 
     def apply_locale(self, locale, out_fn=None):
-        if not out_fn:
```

Comment:
This `out_fn` should stay. The location of the config file is user-overridable, and we shouldn't just ignore that.

Reply:
`out_fn` isn't used and it isn't configurable in systemd AFAIK, so ignoring it make sense.

Reply:
I'm saying it's user configurable via https://cloudinit.readthedocs.io/en/latest/topics/modules.html?highlight=locale#locale

If the user provides us an option, even if it's completely invalid, we don't just ignore it. If no other value other than '/etc/locale.gen' is valid, then check that they've passed '/etc/locale.gen'. You could log a warning and write anyway, or you could raise an exception, but we shouldn't just ignore user input.

Reply:
Oh. I added a warning.

Reply:
You should use `LOG.warning` for the log instead of `util.logexc`. `util.logexc` is meant to be used in an except block and will gather exception information.

Reply:
Thanks. I got a bit confused so I just guessed :)